agent/agent.go

```diff
@@ -665,6 +677,17 @@ func (a *Agent) runPreAndMain(ctx context.Context, tc *taskContext) (status stri
 	}
 	go a.startTimeoutWatcher(timeoutWatcherCtx, execTimeoutCancel, timeoutOpts)
 
+	tc.logger.Execution().Infof("Setting heartbeat timeout to type '%s'.", execTimeout)
```

Comment:
I intentionally put the set/reset for the heartbeat timeout in the foreground rather than absorb it into `startTimeoutWatcher` goroutine because if it runs in the background, the heartbeat set/reset could race between background timers (and set the wrong timeout value).